helper/runner.go

```diff
@@ -135,6 +135,11 @@ func (r *Runner) WalkResources(resourceType string, walker func(*terraform.Resou
 	return nil
 }
 
+// WalkModuleCalls visits all module calls from Files.
+func (r *Runner) WalkModuleCalls(walker func(*terraform.ModuleCall) error) error {
+	return nil
```

Comment:
I've left this blank because I'm just not sure what to implement for this mock. The only caller I could identify is the `runner_test` here, which IIUC is just validating the mock itself functions. Is the idea to bring over some subset of https://github.com/hashicorp/terraform/blob/v0.12.26/configs/module_call.go#L33 ?

Reply:
Yes. This is a helper that allows you to write tests for `ModuleCalls()` when building your plugin.
For instance, this is an example of `Backend()`.

https://github.com/terraform-linters/tflint-ruleset-template/blob/e7d6bf38e5271f6bd685a8171124c0e2c294a916/rules/terraform_backend_type_test.go
https://github.com/terraform-linters/tflint-plugin-sdk/pull/49

I don't think it's a good design, but now I re-implemented it in a simple way in order to avoid dependency of the SDK on hashicorp/terraform.

Reply:
I'd completely overlooked that feature! It's actually going to be pretty handy for the ruleset I'm writing. I just pushed up a change that implements a passable version of module call decoding.

I've implemented a rule+test in the template repo that confirms I can at least assert things about source/version/providers, which seem like the highest value linting targets from my perspective. I'll try to open a PR for that later today.